src/main/javascript/js/restFrontend.js

```diff
@@ -157,7 +157,7 @@ var HTMLchanger = (function () {
         });
     });
 
-    $(document).on('click', '.delete', function (event) {
+    var deleteFood = $(document).on('click', '.delete', function (event) {
         var foodId = $(event.target).parent().find('.foodName').attr('id');
         $.ajax({
             url: 'http://localhost:8080/food/' + foodId,
```

Comment:
create function for request in another file. You have a lot of host duplicates and data, contentTye.